src/lib/modules/contracts_manager/index.js

```diff
@@ -118,11 +118,46 @@ class ContractsManager {
             try {
               const gas = await contractObj.methods[req.body.method].apply(this, req.body.inputs).estimateGas();
               contractObj.methods[req.body.method].apply(this, req.body.inputs)[funcCall]({from: account, gasPrice: req.body.gasPrice, gas: Math.floor(gas)}, (error, result) => {
+                const paramString = abi.inputs.map((input, idx) => {
+                  const quote = input.type.indexOf("int") === -1 ? '"' : '';
+                  return quote + req.body.inputs[idx] + quote;
+                }).join(', ');
+
+                let contractLog = {
+                  name: req.body.contractName,
+                  functionName: req.body.method,
+                  paramString: paramString,
+                  address: contract.deployedAddress,
+                  status: '0x0'
+                };
+
                 if (error) {
+                  self.events.emit('contracts:log', contractLog);
                   return res.send({result: error.message});
                 }
 
-                res.send({result});
+                if(funcCall === 'call') {
+                  contractLog.status = '0x1';
+                  self.events.emit('contracts:log', contractLog);
+                  return res.send({result});
+                }
+
+                self.events.request("blockchain:object", blockchainConnector => {
+                  const {web3} = blockchainConnector;
+
+                  web3.eth.getTransaction(result, (err, tx) => {
+                    contractLog = Object.assign(contractLog, {
+                      data: tx.input,
+                      status: '0x1',
```

Comment:
Again, might be able to get the status from the `tx`

Reply:
A transaction doesn't have a status, a receipt does. The receipt, however, doesn't have the data, but the transaction does.

We assume `0x1` here otherwise it wouldn't have been committed to the blockchain. Am I wrong?